pkg/pillar/pubsub/doc.go

```diff
@@ -50,13 +50,13 @@
 //
 // The driver is responsible for implementing the underlying mechanics of
 // publishing and subscribing. While `pubsub.PubSub` and its components -
-// `Publication` and `Subscription` - handle the in-memory and diff aspects,
+// `NewPublication` and `NewSubscription` - handle the in-memory and diff aspects,
 // the driver itself is responsible for communicating between the publisher
 // and subscriber, and performing any persistence.
 //
 // The driver is expected to implement the `Driver` interface, which primarily
 // involves being able to return the `DriverPublisher` and `DriverSubscriber`.
-// These in turn are used by `Publication` and `Subscription` to publish and
+// These in turn are used by `NewPublication` and `NewSubscription` to publish and
```

Comment:
Again, revert this change. It is the `Publication` and `Subscription` that do the publishing and subscribing; `NewSub` and `NewPub` are just the func calls to create them.